File: examples/Reduce.py
#!/usr/bin/env python3

# pylint: disable = bad-whitespace
# pylint: disable = eval-used
# pylint: disable = invalid-name
# pylint: disable = missing-docstring

# ---------
# Reduce.py
# ---------

# https://docs.python.org/3.6/library/functools.html

from functools import reduce
from operator  import add, mul, sub

def test1 () :
    assert reduce(add,  [2, 3, 4], 0) == 9

def test2 () :
    assert reduce(mul,  (2, 3, 4), 1) == 24

def test3 () :
    assert reduce(sub,  {2, 3, 4}, 2) == -7

def test4 () :
    assert reduce(None, [],        3) == 3

def main () :
    for n in range(4) :
        eval("test" + str(n + 1) + "()")

if __name__ == "__main__" : # pragma: no cover
    print("Reduce.py")
    main()
    print("Done.")

File: examples/Reduce.js
#!/usr/bin/env node

/* jshint
    esversion: 6,
    evil:      true
*/

// ---------
// Reduce.js
// ---------

// https://www.w3schools.com/jsref/jsref_reduce.asp

"use strict";

const assert = require('assert');
const _      = require('lodash');

function test1 () {
    assert(_.reduce([2, 3, 4], _.add,      0) == 9);}

function test2 () {
    assert(_.reduce([2, 3, 4], _.multiply, 1) == 24);}

function test3 () {
    assert(_.reduce([2, 3, 4], _.subtract, 2) == -7);}

function test4 () {
    assert(_.reduce([],        null,       3) == 3);}

function main () {
    for (let i of _.range(4))
        eval("test" + (i + 1) + "()");}

console.log("Reduce.js");
main();
console.log("Done.");
